crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -101,13 +101,23 @@ where
         match data_type {
             #[cfg(feature = "dtype-categorical")]
             DataType::Categorical(rev_map, ordering) => {
-                polars_ensure!(
-                    self.dtype() == &DataType::UInt32,
-                    ComputeError: "cannot cast numeric types to 'Categorical'"
-                );
-                // SAFETY
-                // we are guarded by the type system
-                let ca = unsafe { &*(self as *const ChunkedArray<T> as *const UInt32Chunked) };
+                let ca = match self.dtype() {
+                    DataType::UInt32 => {
+                        // SAFETY: we are guarded by the type system
+                        unsafe { &*(self as *const ChunkedArray<T> as *const UInt32Chunked) }
+                            .clone()
+                    },
+                    DataType::Int8
```

Comment:
Shouldn't this only be allowed for `Enum`? 

Do we need to bound check or can underlying categories be OOB? Not sure without looking it up.

Reply:
Yes, for other categoricals it does not really make sense. We do the bounds check for enums, for the globals we currently don't (legacy). 

The from_global_indices seems to me like a hack, we should always have the rev_map when casting from `UInt32`  to `Categorical` and check for it.